__tests__/unit/http_adapter.test.ts

```diff
@@ -172,7 +173,8 @@ describe('HttpAdapter', () => {
     })
 
     describe('request compression', () => {
-      it('sends a compressed request if compress_request: true', async () => {
+      // FIXME: this one hangs for an unknown reason
+      it.skip('sends a compressed request if compress_request: true', async () => {
```

Comment:
This is the only failing test so far.

Reply:
Fixed, the issue was that I forgot to emit `socket` event that actually triggers the stream pipeline.